code-challenges.test.js

```diff
@@ -26,26 +30,194 @@ const colors2 = [
+// FAIL  ./code-challenges.test.js
+// arrayContaining
+//   ✕ takes in an array and removes the first item from the array and shuffles the remaining content (1 ms)
+
+// ● arrayContaining › takes in an array and removes the first item from the array and shuffles the remaining content
+
+//   ReferenceError: arrayContaining is not defined
 
 // b) Create the function that makes the test pass.
 
+// Process: GREEN!
+
+// PSEUDOCODE:
+// Create a function that takes in an array, removes the first item from the array and shuffles the remaining content.
+
+// Input: an array ("takes in..")
+// Output: removes the first item from the array and shuffles the remaining content
+
+// Given arrays:
+// Input: ["purple", "blue", "green", "yellow", "pink"]
+// Output: ["yellow", "blue", "pink", "green"]
+
+// Input: [ "chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+// Output: ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+
+// Create a function named "arrayContaining" that takes in an array named "array". 
+// Create a function named "newArray" that removes the first item from the array using the .slice() method to remove the first item from the array. Leave the parameter of the .slice method empty since the array is going to be shuffled. 
+// To begin to shuffle the array, create a for loop and let the index equal to the length of the array minus one in order to start at the last index, define the condition of the for loop to let the index be more than zero (so the index will start at index 1) and then decrease a value of the index by 1 each time.  
+// Create a function named "shuffle" that uses the math logic Math.floor and Math.random with no parameters to randomly select an index starting at zero index and then we need to multiply the index plus 1 (since we want our array to be more than zero and begin at index 1. 
+// Now that you have set the code and math logic for the two functions "newArray" and "shuffle", you can create a new array to contain the newArray with any index value together with the newArray shuffle index to equal the newArray shuffle index together with the newArray with any index value (so it is constantly shuffling the array).  
+// Return the newArray containing the values that removes the first item from the array and shuffles the remaining content.
+
+const arrayContaining = (array) => {
+  const newArray = array.slice()
+    for (let i = array.length -1; i > 0; i--) {
+  const shuffle = Math.floor(Math.random() * (i + 1));
+      [newArray[i], newArray[shuffle]] = [newArray[shuffle], newArray[i]] 
+    } 
+    return newArray
+}
+
+// PASSED!
+// PASS  ./code-challenges.test.js
+// arrayContaining
+//   ✓ takes in an array and removes the first item from the array and shuffles the remaining content (1 ms)
+
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
+// Snapshots:   0 total
+// Time:        0.145 s, estimated 1 s
+
+
```

Comment:
Your thought process is great, there's just a couple things to note. 1) in your test, you are testing the variables `colors1` and `colors2` instead of the function call with the variables as arguments. This is causing the test to pass even though the function is not actually doing what you want it to. 2) The shuffle method is great, but look into `.slice()` again. The way it's set up right now, your variable `newArray` is the same array as the original array that is being passed in!